### PR TITLE
RFC: Use block indent style for struct tuple

### LINKED ISSUE #1367: Formatting of multi-line tuple structs doesn't really work
This line overflows:

```
let item_budget = try_opt!(context.config.max_width.checked_sub(item_indent.width() + 2));
```

Because `item_indent` includes the length of the tuple struct up to the `(`, so if that is more than one line, then we are quit.

e.g.

```
 struct SomeStruct<
     ExcessivelyLongGenericName,
     ExcessivelyLongGenericName,
     AnotherExcessivelyLongGenericName,
 > (
     Vec<ExcessivelyLongGenericName>,
     Vec<AnotherExcessivelyLongGenericName>,
 ) where T: Foo ;
```

### PR DESCRIPTION
**Pull request by topecongiro**
Closes #1367.

### CODE REVIEW

File: src/expr.rs [hunk 1]
```diff
@@ -2937,3 +2943,14 @@ impl<'a> ToExpr for TuplePatField<'a> {
...
     }
 }
+
+impl<'a> ToExpr for ast::StructField {
+    fn to_expr(&self) -> Option<&ast::Expr> {
+        None
+    }
+
+    #[allow(unused_variables)]
+    fn can_be_overflowed(&self, context: &RewriteContext, len: usize) -> bool {
```
Comment (nrc) [inline src/expr.rs]: I'd prefer just use `_` for the variables, rather than use the attribute here

Comment (topecongiro) [inline src/expr.rs]: I totally agree. Thanks!

**Comment by nrc** [comment]
Sorry, needs a rebase

**Comment by topecongiro** [comment]
Rebased.

**Pull request merged**